code/game/objects/effects/glowshroom.dm

```diff
@@ -41,38 +32,21 @@
 	icon_state = "shadowshroom"
 	myseed = /obj/item/seeds/glowshroom/shadowshroom
 
-/obj/structure/glowshroom/single/Spread()
-	return
-
-/obj/structure/glowshroom/examine(mob/user)
-	. = ..()
-	. += "This is a [generation]\th generation [name]!"
-
 /**
  * Creates a new glowshroom structure.
  *
  * Arguments:
  * * newseed - Seed of the shroom
  * * mutate_stats - If the plant needs to mutate their stats
- * * spread - If the plant is a result of spreading, reduce its stats
  */
-
-/obj/structure/glowshroom/Initialize(mapload, obj/item/seeds/newseed, mutate_stats, spread)
+/obj/structure/glowshroom/Initialize(mapload, obj/item/seeds/newseed)
 	. = ..()
 	if(newseed)
 		myseed = newseed.Copy()
 		myseed.forceMove(src)
 	else
 		myseed = new myseed(src)
-	if(spread)
-		myseed.adjust_potency(-round(myseed.potency * 0.25)) // Reduce potency of the little mushie if it's spreading
-	if(mutate_stats) //baby mushrooms have different stats :3
-		myseed.adjust_potency(rand(-4,3))
-		myseed.adjust_yield(rand(-3,2))
-		myseed.adjust_production(rand(-3,3))
-		myseed.adjust_endurance(myseed.endurance + rand(-3,2))
-	if(myseed.production >= 1) //In case production is varedited to -1 or less which would cause unlimited or negative delay.
-		delay_spread = delay_spread - (11 - myseed.production) * 100 //Because lower production speed stat gives faster production speed, which should give faster mushroom spread. Range 200-1100 deciseconds.
```

Comment:
Clamp these stats, and it fixes these issues. Every time the shrooms spread, the init proc has mutate_stats=TRUE. Since the values arent clamped, you'll get some stupid and unexpected values breaking the rest of the logic. For instance, the way the yield loop is set up it will still run on negative yield, so a yield of -30 would attempt to make 32 new mushrooms with yield of -30 with a rand(-3,2) modifier. The negative value also breaks the yield based spread termination.

Reply:
I have no intent of fixing the issues **again**. Oranges said a bit of 3 months ago when he last fixed Glowshrooms that if something like this happened again they'd be removed. I am tired of the constant ping-pong game of when Glowshrooms next break the game due to their ability to generate large amounts of processing from very limited player inputs, like a DRDoS against the MC where one player can click once and spawn an entity that ends up spawning a hundred timers over its lifetime.

In addition, I don't like the Glowshroom spreading in general. They keep breaking things. But not breaking things in the "Oh, gameplay mechanic goes brrr way" - It does it in the "Master Controller goes brrrr, maintainer has to log on, diagnose and force the shift to end" kind of way. Between myself and Potato on two shifts last night we couldn't get the shift to recover from SStimer issues without disabling timers entirely, which renders the shift completely unplayable.

They're divisive. They're overused. I've spoken to @coiax about plans for something different and they're willing to implement something new ala Conway's Game of Life as a spreading mechanic, or alternate methods to spread this complete eye cancer across the station.

Reply:
If we remove everything with bugs, there wont be anything left to play with 🙂

Reply:
Make an exception. Just this once for something exceptionally awful :)))